bin/check-ssl-qualys.rb

```diff
@@ -82,17 +90,31 @@ class CheckSSLQualys < Sensu::Plugin::Check::CLI
   option :between_checks,
          short: '-t SECONDS',
          long: '--time-between SECONDS',
-         description: 'The time between each poll of the API',
+         description: 'The fallback time between each poll of the API, when an ETA is given by the previous response and is higher than this value it is used',
          proc: proc { |t| t.to_i },
          default: 10
 
+  option :timeout,
+         short: '-t SECONDS',
+         descriptions: 'the ammount of seconds that this is allowed to run for',
+         proc: proc(&:to_i),
+         default: 300
+
   def ssl_api_request(from_cache)
     params = { host: config[:domain] }
-    params[:startNew] = 'on' unless from_cache
+    params[:startNew] = if from_cache == true
+                          'off'
+                        else
+                          'on'
+                        end
```

Comment:
I'm not sure if passing `startNew=off` is needed.  The v3 api doc is vague but it seems to suggests the parameter be omitted completely for some reason.

> startNew - if set to "on" then cached assessment results are ignored and a new assessment is started. However, if  there's already an assessment in progress, its status is delivered instead. This parameter should be used only once to initiate a new assessment; further invocations should omit it to avoid causing an assessment loop.

https://github.com/ssllabs/ssllabs-scan/blob/master/ssllabs-api-docs-v3.md#invoke-assessment-and-check-progress

This seems a bit odd -- perhaps we should ask for clarification?

Reply:
That feels pretty straightforward the first request should be with it off and the second one should be on. This assumes that you do not want cached or in-progress results. This keeps the old behavior and I am not sure the value of checking for a new report on a somewhat regular basis if you are gonna take cached results. Maybe we can expose that as a param so the user can decide whether they want it be sure that it has just been initiated. The reason I think that the current implementation makes sense is that say a report was in progress and you make a change to fix it you will not get the new results.

Reply:
I suppose we could have an option to control if all requests should ask for cached.